src/lib.rs

```diff
@@ -61,9 +79,38 @@ impl SecretClient {
     ) -> Result<SecretClient> {
         let token = match token {
             Some(token) => token,
-            None => read_vault_token(TOKEN_PATH)?,
+            None => read_token_from(VAULT_TOKEN_PATH)?,
         };
 
+        Self::create_internal(address, mount, base_path, &token)
+    }
+
+    /// Convenience method to create a new SecretClient with a login to vault.
+    ///
+    /// - address is the address of your Vault instance
+    /// - auth_mount is the mount path of the vault authentication
+    /// - role is the vault role to use for the login
+    /// - mount is the mount point of the KV2 secrets engine
+    /// - base_path reflects the lowest level of where secrets are located
+    #[cfg(feature = "auth")]
+    pub async fn create(
```

Comment:
Let's discuss naming of this function. The static one created with token is called `::new` what is fine I guess. this one logs in to vault and authorizes. So `SecretClient::from_auth`, `::from_login`, `::auth`?

Any ideas? /cc @nelusnegur @JPisCoding @adrobisch

Reply:
From offline discussion: `SecretClient::authenticate`?

Reply:
`from_login` sounds the most *rusty* to me and should also be extensible enough if `vaultier` wants to support more than k8s in the future.

Reply:
A bit late to the party. My 2 cents: 

I'd go with `login`, but also make it more obvious that this seems to use the Kubernetes Auth method under the hood (AFAICS). So something like `login_with_k8s`. Haven't looked into the details much, though.

`create` is a nice name, but quite generic for this, IMO.